Add csv output option and domain selection to deku-benchmark

InfiniteSwerve opened the pull request:
## Problem

We want to be able to write benchmark data to csv files, and choose the amount of domains from the command line
## Solution

This PR uses cmdliner to allow optional writing to a csv file, with a mandatory domain choice

d4hines commented on output.csv:
```diff
@@ -0,0 +1,70 @@
+produce, 1000000, 0.489075
```
Please delete this file.

d4hines commented on deku-p/src/core/bin/benchmark/deku_benchmark.ml:
```diff
@@ -267,15 +267,47 @@ let benches : (module BENCH) list =
+  Format.eprintf "%s%!" data
+
+type params = {
+  domains : int; [@env "DEKU_DOMAINS"]
```
```suggestion
  domains : int; [@env "DEKU_DOMAINS"][@default 16]
```

d4hines replied:
Also please delete the code starting on line 8 that defines `domains`.

InfiniteSwerve replied:
Done, good catches.

d4hines reviewed (changes requested):
See comments. Addiotionally can you:
- print a header row, something like: `benchmark, items, duration`. You might want to delete the file if it exists already.
- make the `items` column a number only - no text.

InfiniteSwerve commented:
@d4hines Branch has been rebased and I've modified the code based on your comments. If that's all and there's nothing else blocking, then please approve and merge.

InfiniteSwerve commented:
@d4hines I've rebased and addressed your comments with the latest commit. The changes made to the writer function work, but I'm not sure whether I've made things too complex. If there are no other blocking issues then please merge this PR.

d4hines reviewed (approved):
:+1: 
[all might happy with progress](https://media.giphy.com/media/UJv4aOufZ3INq/giphy.gif)
